### PR TITLE
Adds String subscripting

### PR DESCRIPTION
**Pull request by 3lvis**
I'm not sure if this is really smart or really dumb but damn, it's useful.

- [x] Add unit tests

### CODE REVIEW

**Comment by elland** [comment]
Use case? 🤔

**Comment by 3lvis** [comment]
Need to get the initials. Basically in my just case is just to get the first letter.

screen shot 2016-11-17 at 15 17 52

**Comment by elland** [comment]
That seems like too specific a case.

**Comment by elland** [comment]
Not super convinced the behaviour should be to return an empty string if out of range. I think this is a great case for optionals.

File: Tests/Tests.swift [hunk 1]
```diff
@@ -59,6 +59,11 @@ class Tests: XCTestCase {
         expected = "😍"
         XCTAssertEqual(result, expected)
 
+        string = "👯try longer😍"
+        result = string[11]
+        expected = "😍"
+        XCTAssertEqual(result, expected)
```
Comment (3lvis) [inline Tests/Tests.swift:65]: What's the need of this test? Anything you're looking for?

Comment (elland) [reply Tests/Tests.swift:65]: Checking that even if you mix and match emoji and regular text that the indices are still correct. Actually this test is incomplete. Adding another.

**Comment by elland** [comment]
#18

**Comment by 3lvis** [comment]
Is this good to go then?

**Comment by 3lvis** [comment]
CI is working! 🎉

**Comment by elland** [comment]
I only fixed or tested the first case, as a way to show you how it should be done to account for 32bit wide emoji, and left the other methods for you to check. 👍

**Comment by designatednerd** [comment]
This is now in the standard lib in swift 4 - should we close this PR?

**Pull request closed without merging**